__tests__/i18n.test.ts

```diff
@@ -0,0 +1,34 @@
+import { LanguageKeys } from '@lib/types/Languages';
+import { createLanguageStore } from '@mocks/MockLanguageStore';
+import English from '@root/languages/en-US';
+import Spanish from '@root/languages/es-ES';
+import { client } from '@testlib/MockInstances';
+
+/**
+ * Voodoo magic function that flattens an object, no matter the amount of deep-level children a key has into an array
+ * So it can be tested for strict equality and order of keys.
+ * Taken from Stackoverflow
+ * @param obj Object to deep reduce to keys
+ * @param prefix Prefix to apply to each entry
+ */
+const i18nReducer = (obj: any, prefix = ''): any => Object.keys(obj).reduce((res: string | any[], el: string) => {
+	if (Array.isArray(obj[el])) {
+		return res;
+	} else if (typeof obj[el] === 'object' && obj[el] !== null) {
+		return [...res, ...i18nReducer(obj[el], `${prefix}${el}.`)];
+	}
+	return [...res, prefix + el];
+}, []);
```

Comment:
This looks a lot like Klasa's `objectToTuples`, except far more compute expensive (`@klasa/utils`'s `objectToTuples` iterates only once, this one does Object.keys, reduces, and concatenates, which are 3 iterations).

Maybe adapt the klasa code? https://github.com/dirigeants/utils/blob/master/src/lib/objectToTuples.ts